clients/fides-js/src/fides-tcf.ts

```diff
@@ -184,11 +184,18 @@ const init = async (config: FidesConfig) => {
       config.options.debug,
       "Explicit fidesString detected. Proceeding to override all TCF preferences with given fidesString"
     );
-    cookie.fides_string = config.options.fidesString;
-    cookie.tcf_consent = transformFidesStringToCookieKeys(
+    const { cookieKeys, success } = transformFidesStringToCookieKeys(
       config.options.fidesString,
       config.options.debug
     );
+    if (!success) {
+      // Treat an unsuccessful transform as if no fidesString were passed in.
+      // eslint-disable-next-line no-param-reassign
+      config.options.fidesString = null;
+    } else {
+      cookie.tcf_consent = cookieKeys;
+      cookie.fides_string = config.options.fidesString;
+    }
```

Comment:
Don't we need a similar type of fallback if we fail to decode the data from the `fides_consent` cookie or from the `experience`?

Reply:
this was the only place I found that does a call to `TCString.decode`—I don't think we directly parse strings from the cookie or from the experience (we typically use the `cookie.tcf_consent` obj instead)

Reply:
OK 👍 

I think that'll change pretty quickly, I'm planning to have us load `fides_string`s from API calls promptly. So I think we should plan for that and just refactor this a bit... how about this:
```
if (config.options.fidesString) {
  updateFidesCookieFromString(cookie, config.options.fidesString);
}
```

...and then encapsulate the error handling in that helper